numba_rvsdg/core/datastructures/basic_block.py

```diff
@@ -146,15 +169,23 @@ class SyntheticExitBranch(SyntheticBranch):
 @dataclass(frozen=True)
 class RegionBlock(BasicBlock):
     kind: str = None
-    headers: Dict[str, BasicBlock] = None
-    """The header of the region"""
+    """The kind of region. Can be 'head', 'tail', 'branch', 
+    'loop' or 'meta' strings"""
+    parent_region: "RegionBlock" = None
+    """The parent region of this region as per the SCFG."""
+    header: str = None
+    """The header node of the region"""
     subregion: "SCFG" = None
-    """The subgraph excluding the headers
-    """
+    """The subgraph as an independent SCFG. Note that in case
+    of subregions the exiting node may point to blocks outside
+    of the current SCFG object."""
     exiting: str = None
-    """The exiting node.
-    """
+    """The exiting node of the region."""
+
+    def replace_header(self, new_header):
+        """Does inplace replacement of the header block."""
+        object.__setattr__(self, "header", new_header)
 
-    def get_full_graph(self):
-        graph = ChainMap(self.subregion.graph, self.headers)
-        return graph
+    def replace_exiting(self, new_exiting):
+        """Does, inplace replacement of the exiting block."""
+        object.__setattr__(self, "exiting", new_exiting)
```

Comment:
Same here

Reply:
Same logic as https://github.com/numba/numba-rvsdg/pull/61#discussion_r1210273251 but for exiting blocks